### ISSUE TITLE
Cannot load admin page on admin account

### ISSUE THREAD

**Issue by coatmaker618** [friendica/friendica#8681]
When I login through the web on the admin account I am presented with the option to access the Admin page on the dropdown menu, as expected.

However, when I click on that the page fails to load "ERR_EMPTY_RESPONSE"

Checking the logs (in the friendica/logs folder) I see this line repeated 45 times with different IDs

2020-05-24T01:04:25Z index [NOTICE]: no data for session {"session_id":"n16ig2el2ggo9l894imbko5duo","uri":"\/"} - {"file":"Database.php","line":72,"function":"read","uid":"74dc0e","process_id":8444}

I'm using a standalone MariaDB for the database that seems to work fine (after all I can login from it).

Any thoughts on why I'm presented with the Admin page on the dropdown but it does not load when I click on it?

**Comment by annando** [comment]
Is the page completely blank? if yes, please check your PHP error log.

**Comment by coamtaker618** [comment]
It's not even blank, it fails to load :(  I will see what I can find regarding php error log!

**Comment by coamtaker618** [comment]
Update.  Not sure where to go to find php logs >_<

https://stackoverflow.com/questions/5127838/where-does-php-store-the-error-log-php5-apache-fastcgi-cpanel


Gives 2 locations:
1) /var/log/httpd/ <-- does not exist for me
2) /var/log/apache2/error.log <-- does not seem to change when I try to access this page. Lots of previous data I can share

$ php --info | grep error
```
display_errors => Off => Off
display_startup_errors => Off => Off
error_append_string => no value => no value
error_log => no value => no value
error_prepend_string => no value => no value
error_reporting => 22527 => 22527
html_errors => Off => Off
ignore_repeated_errors => Off => Off
log_errors => On => On
log_errors_max_len => 1024 => 1024
track_errors => Off => Off
xmlrpc_error_number => 0 => 0
xmlrpc_errors => Off => Off
intl.error_level => 0 => 0
opcache.error_log => no value => no value
```

**Comment by MrPetovan** [comment]
Thanks, the connection to your database seems unstable, which has unpredictable results on Friendica.

Are you sure your database server has enough ressources (memory limits, mostly)?

**Comment by coamtaker618** [comment]
Oh, I should have made a disclaimer about that.  I was doing some testing with firewalls.  So the database connection was indeed cutting out during that....but that was before these issues started.  Friendica has since been rebooted (so I would assume that problem would not be "sticky"  Could that assumption be incorrect?)

**Comment by MrPetovan** [comment]
Fair enough, I’m working with what you provide!

I usually clear the error.log when I fixed an issue I know generated a bunch of log messages that are likely to confuse me later.

Then the error you’re getting likely is specific to the FastCGI install which I’m not familiar with, sorry.

**Comment by coamtaker618** [comment]
> Fair enough, I’m working with what you provide!

Of course.  I appreciate the help you were able to provide :)
 
> I usually clear the error.log when I fixed an issue I know generated a bunch of log messages that are likely to confuse me later.

Had I been paying attention to the logs, I surely would have :p
 
> Then the error you’re getting likely is specific to the FastCGI install which I’m not familiar with, sorry.

What is this "FastCGI" install?  Is there a different installation approach that might be worth trying?  And does this still hold if it was previously working?

**Comment by MrPetovan** [comment]
I’m used to the Apache 2 PHP module install, and empty responses are not among the error states of this kind of install. The most likely is a HTTP 500 error when PHP unexpectedly quit, but it is a non-empty response.

I’m mentioning FastCGI because that’s what the Stack Overflow question is about but maybe it has nothing to do with your own install.

**Comment by coamtaker618** [comment]
It could be? I'm not familiar with FastCGI....  but this is sounding like there's a more reliable way to install, or even a step that I'm missing?

If there is a bad state somewhere then my instinct is to reinstall/upgrade.  Is that just stop apache, move old folder, git clone to expected directory, move config back, restart apache---or is there a more detailed walkthrough? Or is there a better approach?

**Comment by MrPetovan** [comment]
Let's just take a step back:
- Is any other Friendica page loading?
- Is any other Friendica page loading using HTTPS?

**Comment by coamtaker618** [comment]
Ok. Yes, as far as I can tell all other pages are loading fine (with HTTPS).  It's just the admin page that's not loading.

**Comment by copiis** [comment]
Did you activate the start page addon? If so please deactivate and try again to load the admin page.

**Comment by coamtaker618** [comment]
"the start page addon?"  Not sure what you mean by that @copiis  (which is to say it's entirely possible that I did and I'm just not remembering).

Also, my approach for acitvating/deactivating addons is through the GUI on the admin page..... how would I do it otherwise?

**Comment by copiis** [comment]
But you still come to the settings page? If so, have a look under Addons if you have the "Startpage" addon. If so, check whether something is entered and in what form.

**Comment by MrPetovan** [comment]
You can alter the site config without access to the GUI if you have access to the command line.

What does the command `php bin/console config system addon` output when ran in the Friendica folder?

**Comment by coamtaker618** [comment]
@copiis thanks, I didn't realize that page was a thing... I always looked at admin page to add/remove addons.  List below of Addon Settings:
Blockem
Cat Avatar Settings
FromApp Settings
IRC Settings
Jappix Mini
MathJax
Content Filter (NSFW and more)

@MrPetovan this is what I see:
```
server@Friendica:/var/www/friendica$ php bin/console config

dir=$(cd "${0%[/\\]*}" > /dev/null; pwd)

if [[ -d /proc/cygdrive && $(which php) == $(readlink -n /proc/cygdrive)/* ]]; then
   # We are in Cgywin using Windows php, so the path must be translated
   dir=$(cygpath -m "$dir");
fi

php "${dir}/console.php" "$@"
```

**Comment by MrPetovan** [comment]
Sorry it's either:
```
$ bin/console config system addon
```
or
```
$ php bin/console.php config system addon
```

**Comment by coamtaker618** [comment]
server@Friendica:/var/www/friendica$ php bin/console.php config system addon
system.addon =>
server@Friendica:/var/www/friendica$ bin/console config system addon
system.addon =>

**Comment by MrPetovan** [comment]
Then no addon seems to be enabled on your node and it isn't a problem related to the startpage addon.

**Comment by coamtaker618** [comment]
So I guess this means the above addons are installed but not enabled?

**Comment by MrPetovan** [comment]
Hmm, maybe I'm missing something, if you see the addon settings, it means they're enabled. Also you can see this admin page but not the main admin page?

**Comment by coamtaker618** [comment]
> Also you can see this admin page but not the main admin page?

Apparently?

>If you see the addon settings, it means they're enabled.

So they're enabled in webgui but not CLI? I'm totally confused!

**Comment by MrPetovan** [comment]
No, the command I gave you was probably wrong, it was off the top of my head and very well might be wrong.

Can you please try all the other admin pages so we can definitely rule the summary page in?

- Overview (`/admin`) <= Confirmed not working
- Federation Statistics (`/admin/federation`)
- Site (`/admin/site`)
- Users (`/admin/users`)
- Addons (`/admin/addons`)
- Themes (`/admin/themes`)
- Additional features (`/admin/features`)
- Terms of Service (`/admin/tos`)
- DB updates (`/admin/dbsync`)
- Inspect Deferred Workers (`/admin/queue/deferred`)
- Inspect worker Queue (`/admin/queue`)
- Contact Blocklist (`/admin/blocklist/contact`)
- Server Blocklist (`/admin/blocklist/server`)
- Delete Item (`/admin/item/delete`)
- Logs (`/admin/logs/`)
- View Logs (`/admin/logs/view`)
- PHP Info (`/admin/phpinfo`)
- Item Source (`/admin/item/source`)

**Comment by coamtaker618** [comment]
Hahahaha, that never occured to me @MrPetovan! So EVERY ONE OF THOSE WORKS JUST FINE \o/  They even gave me the sidebar menu with all of those links!  It only seems to be the base admin page that does not load (still).

PS: No worries about the wrong command.  Does reassure me though :)

**Comment by MrPetovan** [comment]
Alright, so I'd venture is has to do with the checks we perform on this page. Would you be able to edit your files manually for further troubleshooting?

**Comment by coamtaker618** [comment]
I don't see why not

**Comment by MrPetovan** [comment]
Well I can but it's out of scope of this issue! 😅

If you're on the `develop` branch, please pull the latest and comment lines 51-187 of `src/Module/Admin/Summary.php`. In PHP files, this means adding `/*` at the start of line 51, and `*/` to line 187 of the file.

If you're on the latest stable, please comment line 49-163.

**Comment by coamtaker618** [comment]
I am on the stable branch.

Commented out those lines, reloaded apache, and the admin page loaded no problem.

**Comment by MrPetovan** [comment]
HA! Progress. Not need to reload Apache, PHP script are parsed just in time so each file change gives instant results.

Let's curtail the issue. Please comment lines 52-163 now. We will re-enabled each admin summary page check to single out the faulty one.

**Comment by coamtaker618** [comment]
Alright, will do!

**Comment by coamtaker618** [comment]
Good news, it looks like it's just the conditional on line 91.  With only this commented out (everything else uncommented) the page loads fine.
```
 // Check server vitality
/*              if (!self::checkSelfHostMeta()) {
                        $well_known = DI::baseUrl()->get() . '/.well-known/host-meta';
                        $warningtext[] = DI::l10n()->t('<a href="%s">%s</a> is not reachable on your system. This is a severe configuration issue that prevents server to server communication. See <a href="%s">the installation page</a> for help.',
                                $well_known, $well_known, DI::baseUrl()->get() . '/help/Install');
                }
*/
```

**Comment by MrPetovan** [comment]
What happens when you query `/.well-known/host-meta` on your node domain/path in your browser?

**Comment by coamtaker618** [comment]
I get an unstyled XML file.  

It starts with:
<XRD xmlns="http://docs.oasis-open.org/ns/xri/xrd-1.0" xmlns:hm="http://host-meta.net/xrd/1.0">


And ends with:
<Property xmlns:mk="http://salmon-protocol.org/ns/magic-key" type="http://salmon-protocol.org/ns/magic-key" mk:key_id="1">


And has an RSA thing which I assume is a public key, but don't want to post until that's confirmed?

**Comment by coamtaker618** [comment]
Is there a way on github to PM you this info?

**Comment by MrPetovan** [comment]
Yes, it's expected, it's the public key of your instance. Then I'm puzzled about what makes PHP crash so hard there's no error message and no Apache HTTP error output either. Maybe something is wrong with curl, but it would make your instance pretty much isolated since we use it for all instance-to-instance communication.

@annando , any idea?
